src/services/itemMembership/service.ts

```diff
@@ -18,6 +19,7 @@ export class ItemMembershipService {
   itemService: ItemService;
   hosts: Hostname[];
   mailer: MailerDecoration;
+  hooks = new HookManager();
```

Comment:
You can narrow the generic type on the hook manager to get precise types on the `data` argument for both pre and post hooks individually. This will also provide types for the event names themselves.

See:

https://github.com/graasp/graasp/blob/36223a3508df54ce6ff5a75c689449ed02b547da/src/services/item/service.ts#L29-L35

See the type definition at

https://github.com/graasp/graasp/blob/36223a3508df54ce6ff5a75c689449ed02b547da/src/utils/hook.ts#L4-L56